python/sglang/srt/entrypoints/openai/serving_chat.py

```diff
@@ -493,9 +493,6 @@ async def _generate_chat_stream(
                         )
                         yield f"data: {chunk.model_dump_json()}\n\n"
 
-                    if not delta:
```

Comment:
why do we remove this block?

Reply:
https://github.com/sgl-project/sglang/pull/7708 partially addresses the issue where finish_reason = stop even when tool_call exists in the response.

However, that logic won't necessarily work when both reasoning parser and tool call parser are enabled. `If not delta` logic in reasoning_parser will skip the for-loop via `continue` and thus skip the `finish_reason_type = tool_call_finish_reason_type` logic. `finish_reason_type` is still `stop` when tool_call exists in the response.
<img width="698" height="418" alt="image" src="https://github.com/user-attachments/assets/f6a4c25d-e3ab-473b-8585-9d6adddee7b1" />

Reply:
Is this for the case when the last delta is empty but we still need to parse finish_type?

Reply:
> Is this for the case when the last delta is empty but we still need to parse finish_type?

Correct. This is the streaming sequence that will cause finish_reason != 'tool_calls' before my fix:
- ....
- delta="something", finish_reason = None, tool_calls = "something"
- delta = "", finish_reason = "stop", tool_calls=None

The bug is b/c the implementation before my change simply skip the `if finish_reason== stop: set finish_reason = tool_calls` logic in `self. _process_tool_call_stream` due to `if not delta: continue`